### DISCUSSION TITLE
How to perform orderby,pagination and filter from user query.

### DISCUSSION THREAD

**Discussion by aakashkag** [prisma/prisma#2680] [Q&A] [answered]
Hi,
I am using Prisma2 + GraphQL+ Nexus 
 Didn't find any end to end example on filter, orderby and pagination
I would like to do orderby on any field. how we can do that. 
Do we need to expose varibable which will capture field where we are want to applu orderby operation.?
Can we do without passing using varible. ?
and how to enable filter,orderby and pagination at model level. I saw example where users are enabling on nested model only.

```ts
const getAllUser = queryField('getAllUser', {
    type: 'User',
    list: true,
    filtering: true,
    ordering: true,
    pagination: true,
    resolve: async (parent, args, ctx) => {
        try {
            const users = await ctx.prisma.user.findMany({});
            return users;
        } catch (error) {
            console.error(error.stack);
            throw error;
        }
    },
});
```

```ts
const User = objectType({
    name: 'User',
    definition(t) {
        t.model.user_id();
        t.model.user_first_name();
        t.model.user_last_name();
        t.model.user_name();
        t.model.user_email();
  }
});


```
Thanks

**Comment by ryands17** [comment, marked as answer]
Hey @aakashkag :wave: 
You would need to create a query for the `User` type as follows:

```ts
const User = objectType({
  name: 'User',
  definition(t) {
    t.model.user_id()
    t.model.user_first_name()
    t.model.user_last_name()
    t.model.user_name()
    t.model.user_email()
  },
})

const Query = queryType({
  definition(t) {
    t.crud.users({
      filtering: true,
      ordering: true,
    })
  },
})
```

**Reply by aakashkag** [reply]
Hi @ryands17 
Thanks for reply. 

I am getting error. 

I added 
```
const userQuery = queryType({
    definition(t) {
        t.crud.users({
            filtering: true,
            ordering: true,
        });
    },
});
```
```
const getAllUser = queryField('getAllUser', {
    type: 'User',
    list: true,
    resolve: async (parent, args, ctx) => {
        try {
            const users = await ctx.prisma.user.findMany({});
            return users;
        } catch (error) {
            console.error(error.stack);
            throw error;
        }
    },
});


```
Query from client:
```
query{
  getAllUser(orderBy: "user_name"){
    user_id
    user_name
  }
}
```

Error:
```
{
  "error": {
    "errors": [
      {
        "message": "Unknown argument \"orderBy\" on field \"getAllUser\" of type \"Query\".",
        "locations": [
          {
            "line": 2,
            "column": 14
          }
        ]
      }
    ]
  }
}
```

**Reply by ryands17** [reply]
@aakashkag You need to remove this:
```
const getAllUser = queryField('getAllUser', {
    type: 'User',
    list: true,
    resolve: async (parent, args, ctx) => {
        try {
            const users = await ctx.prisma.user.findMany({});
            return users;
        } catch (error) {
            console.error(error.stack);
            throw error;
        }
    },
});

```

And as a query, you would need to call this:
```graphql
query {
  users(orderBy: {name: asc}) {
    name
  }
}
```

**Reply by aakashkag** [reply]
Ohh so it's not possible to apply order by on the fly and keep query business logic as it is. ?

**Reply by ryands17** [reply]
Yes @aakashkag that is a limitation of Nexus currently. It's issue is stated [here](https://github.com/graphql-nexus/nexus-schema-plugin-prisma/issues/303) that you can look into.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
